src/dotnet/commands/dotnet-publish/PublishCommand.cs

```diff
@@ -272,33 +316,60 @@ private static void PublishRefs(LibraryExport export, string outputPath, bool de
             }
         }
 
-        private static int PublishHost(ProjectContext context, string outputPath, CommonCompilerOptions compilationOptions)
+        private static int RenamePublishedHost(ProjectContext context, string outputPath, CommonCompilerOptions compilationOptions)
         {
             if (context.TargetFramework.IsDesktop())
             {
                 return 0;
             }
 
-            foreach (var binaryName in Constants.HostBinaryNames)
+            var publishedHostFile = ResolvePublishedHostFile(outputPath);
+            if (publishedHostFile == null)
             {
-                var hostBinaryPath = Path.Combine(AppContext.BaseDirectory, binaryName);
-                if (!File.Exists(hostBinaryPath))
-                {
-                    Reporter.Error.WriteLine($"Cannot find {binaryName} in the dotnet directory.".Red());
-                    return 1;
-                }
+                Reporter.Output.WriteLine($"publish: warning: host executable not available in dependencies, using host for current platform");
```

Comment:
In our testbed we have a lot of tests which don't have the host in their dependency closure yet (and our partners would too). In trying to convert them I got blocked on our method of producing the NETCoreApp package only for the current rid (this breaks down on windows because it tries to restore both x86 and x64).

To get this scenario lit up, I left in the fallback behavior while I work on unblocking the issues with the package and convert the tests ( #2450 ). 

This is entirely a Point in Time issue and I will try to get it in for RC2.